src/main/java/org/jenkinsci/plugins/durabletask/executors/OnceRetentionStrategy.java

```diff
@@ -56,6 +56,18 @@ public OnceRetentionStrategy(int idleMinutes) {
         super(idleMinutes);
     }
 
+    @Override
+    public long check(final AbstractCloudComputer c) {
+        // When the slave is idle we should disable accepting tasks and check to see if it is already trying to
+        // terminate. If it's not already trying to terminate then lets terminate manually.
+        if (c.isIdle()) {
+            done(c);
```

Comment:
We need to bypass the super method as that does not lock correctly when terminating (at least in older versions of Jenkins)

In any case a Once retention strategy should probably ignore the idle minutes once it has been used... if it has not yet been used then it should consider the idle minutes